titus-common/src/main/java/com/netflix/titus/common/model/validator/Validator.java

```diff
@@ -0,0 +1,13 @@
+package com.netflix.titus.common.model.validator;
+
+import java.util.Set;
+
+/**
+ * A Validator determines whether an object of the parameterized type is valid.  If it finds an object to be invalid it
+ * returns a non-empty set of {@link ValidationError}s.
+ *
+ * @param <T> The type of object this Validator validates.
+ */
+public interface Validator<T> {
+    Set<ValidationError> validate(T entity);
```

Comment:
This should be asynchronous API. There are two reasons for that:
* GRPC calls cannot block, as doing this would block the netty thread
* they cannot be easily parallelized/composed to speedup the execution

The `Validator` name is very generic, and already used by other framework. Can we be more specific here?

Looking at this API proposal, I see many commonalities with the existing `EntitySanitizer` framework. Does it make sense, to base both APIs on the same data model? `EntitySanitizer` uses now JavaBean model (`ConstraintViolation`), but we could use `ValidationError` provided here in both places.